### PR TITLE
v2: more tests

### PR DESCRIPTION
**Pull request by roncohen**

### CODE REVIEW

File: beater/v2_handler_test.go [hunk 1]
```diff
@@ -19,17 +19,37 @@ package beater
...
+	var resultmap map[string]interface{}
+	err := json.Unmarshal(body, &resultmap)
+	assert.NoError(t, err, "err", string(body))
+
+	resultName := fmt.Sprintf("approved-stream-result/%s", name)
+	verifyErr := tests.ApproveJson(resultmap, resultName, nil)
+	if verifyErr != nil {
+		assert.Fail(t, fmt.Sprintf("Test %s failed with error: %s", name, verifyErr.Error()))
+	}
+}
```
Comment (simitt) [inline beater/v2_handler_test.go]: Would it make sense to move `approveResultBody` to the `test` package? I think I saw this copy&pasted a few times in former PR already.

File: beater/v2_handler_test.go [hunk 2]
```diff
@@ -81,3 +101,80 @@ func TestRequestDecoderError(t *testing.T) {
...
+				assert.Equal(t, 0, w.Body.Len())
+			} else {
+				body := w.Body.Bytes()
+				approveResultBody(t, "TestRequestIntegration"+test.name, body)
+			}
+		})
+	}
+}
+
+func TestReportingProblemsIntegration(t *testing.T) {
```
Comment (simitt) [inline beater/v2_handler_test.go]: I'd appreciate to get rid of the duplicated code in `TestRequestIntegration` and `TestReportingProblemsIntegration`.

Comment (roncohen) [reply beater/v2_handler_test.go]: good tip!

File: beater/v2_handler_test.go (hunk 1 above)
Comment (roncohen) [reply beater/v2_handler_test.go]: refactored a bit. WDYT?

**Pull request merged**